src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1080,6 +1080,20 @@ protected override bool IsValidPath(string path)
                 }
             }
 
+            // .NET introduced a change where invalid characters are accepted https://learn.microsoft.com/en-us/dotnet/core/compatibility/2.1#path-apis-dont-throw-an-exception-for-invalid-characters
+            // We need to check for invalid characters ourselves
+
+            // Remove drive root first
+            string pathWithoutDriveRoot = path.Substring(Path.GetPathRoot(path).Length);
+
+            foreach (string segment in pathWithoutDriveRoot.Split(Path.DirectorySeparatorChar))
+            {
+                if (segment.IndexOfAny(Path.GetInvalidFileNameChars()) != -1)
```

Comment:
Can you please have a comment here calling out `GetInvalidFileNameChars` is a super set of `GetInvalidPathChars`?